openpype/modules/deadline/repository/custom/plugins/GlobalJobPreLoad.py

```diff
@@ -190,17 +189,23 @@ def inject_openpype_environment(deadlinePlugin):
             raise RuntimeError(msg)
 
         if not os.environ.get("OPENPYPE_MONGO"):
-            print(">>> Missing OPENPYPE_MONGO env var, process won't work")
+            if not job.GetJobEnvironmentKeyValue('OPENPYPE_MONGO'):
+                print(">>> Missing OPENPYPE_MONGO env var, process won't work")
 
         os.environ["AVALON_TIMEOUT"] = "5000"
 
         args_str = subprocess.list2cmdline(args)
         print(">>> Executing: {} {}".format(exe, args_str))
-        process = ProcessUtils.SpawnProcess(
-            exe, args_str, os.path.dirname(exe)
+
+        current_env = {**os.environ}
+        current_env["OPENPYPE_MONGO"] = job.GetJobEnvironmentKeyValue(
```

Comment:
You can just set `os.environ`. It's single time use process and you don't need to worry aby having any affect.
```suggestion
        os.environ["OPENPYPE_MONGO"] = job.GetJobEnvironmentKeyValue(
```